### PR TITLE
Jr2 876 add documentation and whitelist properties

### PR DESCRIPTION
**Pull request by michael-wirth**

### CODE REVIEW

File: infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSink.java [hunk 1]
```diff
@@ -22,6 +23,13 @@ public void register(JobAdvertisement jobAdvertisement) {
...
     public void deregister(JobAdvertisement jobAdvertisement) {
-        avamWebServiceClient.deregister(jobAdvertisement);
+        if (mustSendDeregistratonNotification(jobAdvertisement)) {
+            avamWebServiceClient.deregister(jobAdvertisement);
+        }
+    }
+
+    private boolean mustSendDeregistratonNotification(JobAdvertisement jobAdvertisement) {
+        // send deregistration message only for JobAds registered by JOBROOM or API.
+        return SourceSystem.JOBROOM.equals(jobAdvertisement.getSourceSystem()) || SourceSystem.API.equals(jobAdvertisement.getSourceSystem());
```
Comment (cjpsycos) [inline infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSink.java]: (jobAd.stellennummerAvam != null) && (SourceSystem.JOBROOM.equals(jobAdvertisement.getSourceSystem()) || SourceSystem.API.equals(jobAdvertisement.getSourceSystem()))

File: infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSinkApplication.java [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+import ch.admin.seco.jobs.services.jobadservice.application.ProfileRegistry;
 
 @SpringBootApplication(exclude = {DataSourceAutoConfiguration.class, JpaRepositoriesAutoConfiguration.class, HibernateJpaAutoConfiguration.class})
+@EnableConfigurationProperties(AvamProperties.class)
 public class AvamWebServiceSinkApplication {
 
     public static void main(String[] args) {
-        SpringApplication.run(AvamWebServiceSinkApplication.class, args);
+        SpringApplication app = new SpringApplication(AvamWebServiceSinkApplication.class);
+        app.setDefaultProperties(ImmutableMap.of(DEFAULT_PROFILES_PROPERTY_NAME, ProfileRegistry.AVAM_WSDL_V1));
```
Comment (cjpsycos) [inline infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSinkApplication.java]: AVAM_WSDL_V2

File: infrastructure-messagebroker-ws-avam-source/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamSourceApplication.java [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 
+import ch.admin.seco.jobs.services.jobadservice.application.ProfileRegistry;
+
 @SpringBootApplication(exclude = {DataSourceAutoConfiguration.class, JpaRepositoriesAutoConfiguration.class, HibernateJpaAutoConfiguration.class})
 public class AvamSourceApplication {
 
     public static void main(String[] args) {
-        SpringApplication.run(AvamSourceApplication.class, args);
+        SpringApplication app = new SpringApplication(AvamSourceApplication.class);
+        app.setDefaultProperties(ImmutableMap.of(DEFAULT_PROFILES_PROPERTY_NAME, ProfileRegistry.AVAM_WSDL_V1));
```
Comment (cjpsycos) [inline infrastructure-messagebroker-ws-avam-source/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamSourceApplication.java]: AVAM_WSDL_V2

File: infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSink.java (hunk 1 above)
Comment (michael-wirth) [reply infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSink.java]: done

File: infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSinkApplication.java (hunk 2 above)
Comment (michael-wirth) [reply infrastructure-messagebroker-ws-avam-sink/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamWebServiceSinkApplication.java]: done

File: infrastructure-messagebroker-ws-avam-source/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamSourceApplication.java (hunk 3 above)
Comment (michael-wirth) [reply infrastructure-messagebroker-ws-avam-source/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamSourceApplication.java]: done

**Pull request merged**